lib/actions-oclif/push.ts

```diff
@@ -444,5 +337,74 @@ export const push: CommandDefinition<
 					create an issue.`,
 				);
 		}
-	},
-};
+	}
+
+	async getBuildTarget(appOrDevice: string): Promise<BuildTarget | null> {
+		const {
+			validateApplicationName,
+			validateDotLocalUrl,
+			validateIPAddress,
+		} = await import('../utils/validation');
+
+		// First try the application regex from the api
+		if (validateApplicationName(appOrDevice)) {
+			return BuildTarget.Cloud;
+		}
+
+		if (validateIPAddress(appOrDevice) || validateDotLocalUrl(appOrDevice)) {
+			return BuildTarget.Device;
+		}
+
+		return null;
+	}
+
+	async getAppOwner(sdk: BalenaSDK, appName: string) {
+		const { selectFromList } = await import('../utils/patterns');
```

Comment:
Could you move this down so we can avoid the import altogether if the app name is non-ambiguous

Reply:
This one still stands